src/st_widgets.c

```diff
@@ -454,14 +401,10 @@ boolean ST_MessagesResponder(event_t *ev)
     } // jff 2/26/98 no chat functions if message review is displayed
     else
     {
-        if (M_InputActivated(input_chat_enter))
-        {
-            ch = KEY_ENTER;
-        }
-
         // send a macro
         if (altdown)
         {
+            ch = ev->data1.i;
```

Comment:
Should macros be simplified or removed? Currently, users have to press “T” and then Alt-Number. I suggest just Alt-Number.

Reply:
I see no problem with macros being only available in chat mode.

Reply:
There is a slightly hacky solution - in SDL_TEXTINPUT mode the combination of Alt-1 is not a printable character, so it sends nothing. Currently, to detect macro in chat mode, we read `ev_keup` event.

I think just the Alt-1 combination without chat mode is more convenient. To be honest, I learned how to send a macro only from reading the code.

Reply:
Fine with me either way. I have never ever sent a single chat macro in my entire life. But I am afraid that it clashes with muscle memory. The ALT key was used for strafing in Vanilla Doom and the number keys were used to change weapons. So, chances are that you accidently send a chat macro when switching weapons while strafing in the heat of the combat. 😉